utils/build-toolchain

```diff
@@ -52,11 +52,18 @@ PRESET_FILE_FLAGS=
 PRESET_PREFIX=
 NO_TEST=",no_test"
 USE_OS_RUNTIME=
+CROSS_COMPILE_HOSTS=
 
 case $(uname -s) in
   Darwin)
     SWIFT_PACKAGE=buildbot_osx_package
     OS_SUFFIX=osx
+
+    CROSS_COMPILE_HOSTS=macosx-arm64
```

Comment:
Not sure if I'm grasping your concern correctly -- are you asking whether we should cross compile at all for x86_64 if we are building the compiler on Apple Silicon?

Or is the way I'm assigning the value (assuming the default scenario is building on x86_64) that is confusing?

Reply:
I'm wondering about the assignment itself.  Cross-compiling for x86_64 on ARM64 seems reasonable.  But what if you are building on x86_64?  In that case, you should be cross-compiling for arm64 right?

Reply:
Yes, that is the intent -- this code does not express that very well (since I am explicitly checking for running on arm64 but not for running on x86_64)

The new approach in #60664 seems to have addressed this